src/service/pluginService.ts

```diff
@@ -127,210 +112,147 @@ export class PluginService {
-    notificationCallBack("TOTAL EXTENSIONS : " + missingList.length);
-    notificationCallBack("");
-    notificationCallBack("");
-    let myExt: string = process.argv0;
-    console.log(myExt);
-    let codeLastFolder = "";
-    let codeCliPath = "";
-    if (osType === OsType.Windows) {
-      if (isInsiders) {
-        codeLastFolder = "Code - Insiders";
-        codeCliPath = "bin/code-insiders";
-      } else {
-        codeLastFolder = "Code";
-        codeCliPath = "bin/code";
-      }
-    } else if (osType === OsType.Linux) {
-      if (isInsiders) {
-        codeLastFolder = "code-insiders";
-        codeCliPath = "bin/code-insiders";
-      } else {
-        codeLastFolder = "code";
-        codeCliPath = "bin/code";
-      }
-    } else if (osType === OsType.Mac) {
-      codeLastFolder = "Frameworks";
-      codeCliPath = "Resources/app/bin/code";
-    }
-    myExt =
-      '"' +
-      myExt.substr(0, myExt.lastIndexOf(codeLastFolder)) +
-      codeCliPath +
-      '"';
-    for (let i = 0; i < missingList.length; i++) {
-      const missExt = missingList[i];
-      const name = missExt.publisher + "." + missExt.name;
-      const extensionCli = myExt + " --install-extension " + name;
-      notificationCallBack(extensionCli);
-      try {
-        const installed = await new Promise<boolean>(res => {
-          exec(extensionCli, (err, stdout, stderr) => {
-            if (!stdout && (err || stderr)) {
-              notificationCallBack(err || stderr);
-              res(false);
-            }
-            notificationCallBack(stdout);
-            res(true);
+    return new Promise((resolve, reject) => {
+      const addedExtensions: ExtensionInformation[] = [];
+      let remainingExtensions: ExtensionInformation[] = [...missingExtensions];
+
+      notificationCallBack("TOTAL EXTENSIONS : " + missingExtensions.length);
+      notificationCallBack("");
+      notificationCallBack("");
+      missingExtensions.forEach(async ext => {
+        const name = ext.publisher + "." + ext.name;
+        try {
+          const installed = await new Promise<boolean>(res => {
+            vscode.commands
+              .executeCommand("workbench.extensions.installExtension", name)
+              .then(() => res(true));
           });
-        });
-        if (installed) {
-          notificationCallBack("");
-          notificationCallBack(
-            "EXTENSION : " +
-              (i + 1) +
-              " / " +
-              missingList.length.toString() +
-              " INSTALLED.",
-            true
-          );
-          notificationCallBack("");
-          notificationCallBack("");
-          addedExtensions.push(missExt);
+          if (installed) {
+            remainingExtensions = remainingExtensions.filter(
+              remExt => remExt.name !== ext.name
+            );
+            notificationCallBack("");
+            notificationCallBack(
+              `EXTENSION: ${ext.name} INSTALLED. ${

```

Comment:
We need  how many of total extensions are installed along with the extension name

Like
1  / 10 installed

Reply:
Ok, I have modified it to show the total amount as well as the remaining amount.